server/src/com/cloud/api/auth/DefaultLoginAPIAuthenticatorCmd.java

```diff
@@ -166,7 +166,7 @@ public String authenticate(String command, Map<String, Object[]> params, HttpSes
                     throw new CloudAuthenticationException("Unable to find the domain from the path " + domain);
                 }
                 final UserAccount userAccount = _accountService.getActiveUserAccount(username[0], domainId);
-                if (userAccount == null || !(User.Source.UNKNOWN.equals(userAccount.getSource()) || User.Source.LDAP.equals(userAccount.getSource()))) {
+                if (userAccount != null && User.Source.SAML2 == userAccount.getSource()) {
```

Comment:
Thanks, let's keep this. We can revisit if we've more user type for which this may cause a security issue.